code/modules/xenomorph/xeno_structures.dm

```diff
@@ -452,3 +450,90 @@
+	///Fire rate of the target in ticks
+	firerate = 20
+
+/obj/structure/xeno/resin/xeno_turret/jelly/Initialize(mapload, hivenumber = XENO_HIVE_NORMAL)
+	.=..()
+	ammo = GLOB.ammo_list[/datum/ammo/xeno/larva_jelly]
+	set_light(2, 2, LIGHT_COLOR_PURPLE)
+
+/obj/structure/xeno/resin/xeno_turret/jelly/process()
+	//Turrets regen some HP, every 2 sec
+	.=..()
+	if(world.time > last_scan_time + TURRET_SCAN_FREQUENCY)
+		scanjel()
+		last_scan_time = world.time
+	if(!hostile || hostile.reagents.has_reagent(/datum/reagent/consumable/larvajelly) || !locate(/obj/item/alien_embryo) in hostile)
+		if(last_hostile)
+			set_last_hostile(null)
+		return
+
+/obj/structure/xeno/resin/xeno_turret/jelly/update_overlays()
+	. = ..()
+	if(obj_integrity <= max_integrity / 2)
+		. += image('icons/Xeno/jelturret.dmi', src, "+turret_damage")
+	if(CHECK_BITFIELD(resistance_flags, ON_FIRE))
+		. += image('icons/Xeno/jelturret.dmi', src, "+turret_on_fire")
+
+///Look for the closest human in range and in light of sight. If no human is in range, will look for xenos of other hives
+/obj/structure/xeno/resin/xeno_turret/jelly/get_target()
+	var/distance = range + 0.5 //we add 0.5 so if a potential target is at range, it is accepted by the system
+	var/buffer_distance
+	var/list/turf/path = list()
+	for (var/mob/living/nearby_hostile AS in potential_hostiles)
+		if(!nearby_hostile || nearby_hostile.reagents.has_reagent(/datum/reagent/consumable/larvajelly) || !locate(/obj/item/alien_embryo) in nearby_hostile)
+			continue
+		buffer_distance = get_dist(nearby_hostile, src)
+		if (distance <= buffer_distance) //If we already found a target that's closer
+			continue
+		path = getline(src, nearby_hostile)
+		path -= get_turf(src)
+		if(!path.len) //Can't shoot if it's on the same turf
+			continue
+		var/blocked = FALSE
+		for(var/turf/T AS in path)
+			if(IS_OPAQUE_TURF(T) || T.density && T.throwpass == FALSE)
+				blocked = TRUE
+				break //LoF Broken; stop checking; we can't proceed further.
+
+			for(var/obj/machinery/MA in T)
+				if(MA.opacity || MA.density && MA.throwpass == FALSE)
+					blocked = TRUE
+					break //LoF Broken; stop checking; we can't proceed further.
+
+			for(var/obj/structure/S in T)
+				if(S.opacity || S.density && S.throwpass == FALSE )
+					blocked = TRUE
+					break //LoF Broken; stop checking; we can't proceed further.
+		if(!blocked)
+			distance = buffer_distance
+			. = nearby_hostile
+
+///Return TRUE if a possible target is near
+/obj/structure/xeno/resin/xeno_turret/jelly/proc/scanjel()
+	potential_hostiles.Cut()
+	for (var/mob/living/carbon/human/nearby_human AS in cheap_get_humans_near(src, TURRET_SCAN_RANGE))
+		if(nearby_human.stat == DEAD || nearby_human.reagents.has_reagent(/datum/reagent/consumable/larvajelly) || !locate(/obj/item/alien_embryo) in nearby_human)
+			continue
+		potential_hostiles += nearby_human
```

Comment:
You should just be able to override `scan()` with this code, rather than having a new proc for it.
```suggestion
///Return TRUE if a possible target is near
/obj/structure/xeno/resin/xeno_turret/jelly/scan()
	potential_hostiles.Cut()
	for (var/mob/living/carbon/human/nearby_human AS in cheap_get_humans_near(src, TURRET_SCAN_RANGE))
		if(nearby_human.stat == DEAD || nearby_human.reagents.has_reagent(/datum/reagent/consumable/larvajelly) || !locate(/obj/item/alien_embryo) in nearby_human)
			continue
		potential_hostiles += nearby_human
```